lib/AST/Decl.cpp

```diff
@@ -1137,7 +1137,7 @@ static bool isPolymorphic(const AbstractStorageDecl *storage) {
     return false;
 
   case DeclKind::Protocol:
-    return true;
+    return !storage->getDeclContext()->isExtensionContext();
 
   case DeclKind::Class:
```

Comment:
Good point. For class members checking for `final` should cover both in-class and extension decls, right?